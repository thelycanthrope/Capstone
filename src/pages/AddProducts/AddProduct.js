import React from 'react'
import {Formik,Form,Field,ErrorMessage, useFormikContext } from "formik"
import * as Yup from 'yup'
import './add.css'
import posts from './posts'

import {toast} from 'react-toastify'
import 'react-toastify/dist/ReactToastify.css'
import { Prompt } from 'react-router'
import NavBar from '../../Components/NavBar'


const initialValues={
    pname:"",
	description:"",
	manufacturer:"",
	price:"",
    quantity:"",
    url:"",
    category:""
}

const validationSchema = Yup.object({
    pname:Yup.string().required('Product Name is mandatory field'),
	description:Yup.string().required('Description is mandatory field'),
	manufacturer:Yup.string().required('Manufacturer is mandatory field'),
	price:Yup.string().required('Price is mandatory field') ,
    quantity:Yup.string().required('Quantity is mandatory field'),
    url:Yup.string().required('URL is mandatory field'),
    category:Yup.string().required('Category is Mandetory Field')
})


const PromptIfDirty = () => {
    const formik = useFormikContext();
    return (
      <Prompt
        when={formik.dirty && formik.submitCount === 0}
        message="Are you sure you want to leave? You have with unsaved changes."
      />
    );
  };




toast.configure();

function AddProduct(props) {

    const onSubmit = (value) =>{
     
       posts(value);
       toast.success("Product added successfully!")
       props.history.push('/home');
     }

    return (
        <div className="bg-dark">
            <NavBar/>
            
        <Formik initialValues={initialValues} validationSchema={validationSchema} onSubmit={onSubmit}  >
            <div> <br></br>
            <br></br>
            <br/>
            <h1 className="text-white text-center">Add New Product</h1>
            <br></br>


          
            <Form className=" container  border rounded "> <PromptIfDirty/>

                <div className=" row">
                <div className="col-sm">
                    <label htmlFor="pname" className="form-label text-white d-flex justify-content-center">Product Nme</label> <span style={{color:"red"}}><ErrorMessage name="pname"/></span>
                    <div className="input-group mb-3">
                        <Field type="text" className="form-control text-white bg-dark" id="pname" name="pname" aria-describedby="basic-addon3" />
                    </div>
                </div>

                <div className="col-sm">
                    <label htmlFor="manufacturer" className="form-label text-white d-flex justify-content-center "> Manufacturer</label><span style={{color:"red"}}><ErrorMessage name="manufacturer"/></span>
                    <div className="input-group mb-3">
                        <Field type="text" className="form-control text-white bg-dark" id="manufacturer" name="manufacturer" aria-describedby="basic-addon3" />
                    </div>
                </div>
                </div>

                <div>
                    <label htmlFor="description" className="form-label text-white d-flex justify-content-center">Description for Product</label><span style={{color:"red"}}><ErrorMessage name="description"/></span>
                    <div className="input-group mb-3">
                        <Field type="text" className="form-control text-white bg-dark" id="description" name="description" aria-describedby="basic-addon3" />
                    </div>
                </div>


                    <div className=" row  ">
                <div className=" col-sm ">
                   <label htmlFor="price" className="form-label text-white d-flex justify-content-center" id="ttttt">Price</label>
                   <span style={{color:"red"}}><ErrorMessage name="price"/></span>
                    <div className="input-group mb-3">
                    <span className="input-group-text text-white bg-dark" id="span">&#8377;</span>  <Field type="text" className="form-control text-white bg-dark  " id="price" name="price" aria-describedby="basic-addon3" />
                    </div>
                </div>


                <div className=" col-sm ">
                   <label htmlFor="quantity" className="form-label text-white d-flex justify-content-center">Quantity</label><span style={{color:"red"}}><ErrorMessage name="quantity"/></span>
                    <div className="input-group mb-3 ">
                        <Field type="text" className="form-control text-white bg-dark " id="quantity" name="quantity" aria-describedby="basic-addon3" />
                    </div>
                </div>

                <div className=" col-sm ">
                   <label htmlFor="category" className="form-label text-white d-flex justify-content-center">Category</label><span style={{color:"red"}}><ErrorMessage name="category"/></span>
                    <div className="input-group mb-3 ">
                        <Field type="text" className="form-control text-white bg-dark " id="category" name="category" aria-describedby="basic-addon3" />
                    </div>
                </div>

                </div>

                <div>
                    <label htmlFor="url" className="form-label text-white fs-3 d-flex justify-content-center">Image URL</label><span style={{color:"red"}}><ErrorMessage name="url"/></span>
                    <div className="input-group mb-3">
                        <span className="input-group-text text-white bg-secondary mr-1" id="basic-addon3">https://exampleimage.com/
                        </span>
                        <Field type="text" className="form-control text-white bg-dark" id="url" name="url" aria-describedby="basic-addon3" />
                    </div>
                </div>

                <div className="">
                    <button type="submit" className="btn text-white w-100 mb-2 bg-dark border" >Submit</button>
                </div>
                
            </Form>
            
            <br></br>
            </div>
            
        </Formik >
        <br></br>
        <br></br>
        <br></br>
        </div>
    )
}


export default AddProduct
