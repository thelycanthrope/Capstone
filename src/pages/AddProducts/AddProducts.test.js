
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import AddProduct from './AddProduct';
import toJson from 'enzyme-to-json'


configure({ adapter: new Adapter() });
describe('checking Add prods snapshot', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<AddProduct />);
  });

  it('taking snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})


describe('Add product checking elements', () => {
  let shallowWrapper

    beforeEach(() => {
        shallowWrapper = shallow(<AddProduct/>);
    });

    test('should have h1', () => {
      expect(shallowWrapper.find("h1").render().text()).toEqual("Add New Product")
    })

    test('should have 7 labels', () => {
      expect(shallowWrapper.find('label').length).toEqual(7)
      
    })
    
    
})
