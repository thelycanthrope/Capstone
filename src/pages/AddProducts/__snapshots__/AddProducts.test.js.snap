// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checking Add prods snapshot taking snapshot 1`] = `
<div
  className="bg-dark"
>
  <Formik
    initialValues={
      Object {
        "category": "",
        "description": "",
        "manufacturer": "",
        "pname": "",
        "price": "",
        "quantity": "",
        "url": "",
      }
    }
    onSubmit={[Function]}
    validationSchema={
      ObjectSchema {
        "_blacklist": ReferenceSet {
          "list": Set {},
          "refs": Map {},
        },
        "_excludedEdges": Array [],
        "_mutate": undefined,
        "_nodes": Array [
          "category",
          "url",
          "quantity",
          "price",
          "manufacturer",
          "description",
          "pname",
        ],
        "_sortErrors": [Function],
        "_typeError": [Function],
        "_whitelist": ReferenceSet {
          "list": Set {},
          "refs": Map {},
        },
        "conditions": Array [],
        "deps": Array [],
        "exclusiveTests": Object {},
        "fields": Object {
          "category": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "description": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "manufacturer": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "pname": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "price": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "quantity": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "url": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
        },
        "spec": Object {
          "abortEarly": true,
          "nullable": false,
          "presence": "optional",
          "recursive": true,
          "strict": false,
          "strip": false,
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
        "type": "object",
      }
    }
  >
    <div>
       
      <br />
      <br />
      <br />
      <h1
        className="text-white text-center"
      >
        Add New Product
      </h1>
      <br />
      <Form
        className=" container  border rounded "
      >
         
        <PromptIfDirty />
        <div
          className=" row"
        >
          <div
            className="col-sm"
          >
            <label
              className="form-label text-white d-flex justify-content-center"
              htmlFor="pname"
            >
              Product Nme
            </label>
             
            <span
              style={
                Object {
                  "color": "red",
                }
              }
            >
              <FormikConnect(ErrorMessageImpl)
                name="pname"
              />
            </span>
            <div
              className="input-group mb-3"
            >
              <Field
                aria-describedby="basic-addon3"
                className="form-control text-white bg-dark"
                id="pname"
                name="pname"
                type="text"
              />
            </div>
          </div>
          <div
            className="col-sm"
          >
            <label
              className="form-label text-white d-flex justify-content-center "
              htmlFor="manufacturer"
            >
               Manufacturer
            </label>
            <span
              style={
                Object {
                  "color": "red",
                }
              }
            >
              <FormikConnect(ErrorMessageImpl)
                name="manufacturer"
              />
            </span>
            <div
              className="input-group mb-3"
            >
              <Field
                aria-describedby="basic-addon3"
                className="form-control text-white bg-dark"
                id="manufacturer"
                name="manufacturer"
                type="text"
              />
            </div>
          </div>
        </div>
        <div>
          <label
            className="form-label text-white d-flex justify-content-center"
            htmlFor="description"
          >
            Description for Product
          </label>
          <span
            style={
              Object {
                "color": "red",
              }
            }
          >
            <FormikConnect(ErrorMessageImpl)
              name="description"
            />
          </span>
          <div
            className="input-group mb-3"
          >
            <Field
              aria-describedby="basic-addon3"
              className="form-control text-white bg-dark"
              id="description"
              name="description"
              type="text"
            />
          </div>
        </div>
        <div
          className=" row  "
        >
          <div
            className=" col-sm "
          >
            <label
              className="form-label text-white d-flex justify-content-center"
              htmlFor="price"
              id="ttttt"
            >
              Price
            </label>
            <span
              style={
                Object {
                  "color": "red",
                }
              }
            >
              <FormikConnect(ErrorMessageImpl)
                name="price"
              />
            </span>
            <div
              className="input-group mb-3"
            >
              <span
                className="input-group-text text-white bg-dark"
                id="span"
              >
                ₹
              </span>
                
              <Field
                aria-describedby="basic-addon3"
                className="form-control text-white bg-dark  "
                id="price"
                name="price"
                type="text"
              />
            </div>
          </div>
          <div
            className=" col-sm "
          >
            <label
              className="form-label text-white d-flex justify-content-center"
              htmlFor="quantity"
            >
              Quantity
            </label>
            <span
              style={
                Object {
                  "color": "red",
                }
              }
            >
              <FormikConnect(ErrorMessageImpl)
                name="quantity"
              />
            </span>
            <div
              className="input-group mb-3 "
            >
              <Field
                aria-describedby="basic-addon3"
                className="form-control text-white bg-dark "
                id="quantity"
                name="quantity"
                type="text"
              />
            </div>
          </div>
          <div
            className=" col-sm "
          >
            <label
              className="form-label text-white d-flex justify-content-center"
              htmlFor="category"
            >
              Category
            </label>
            <span
              style={
                Object {
                  "color": "red",
                }
              }
            >
              <FormikConnect(ErrorMessageImpl)
                name="category"
              />
            </span>
            <div
              className="input-group mb-3 "
            >
              <Field
                aria-describedby="basic-addon3"
                className="form-control text-white bg-dark "
                id="category"
                name="category"
                type="text"
              />
            </div>
          </div>
        </div>
        <div>
          <label
            className="form-label text-white fs-3 d-flex justify-content-center"
            htmlFor="url"
          >
            Image URL
          </label>
          <span
            style={
              Object {
                "color": "red",
              }
            }
          >
            <FormikConnect(ErrorMessageImpl)
              name="url"
            />
          </span>
          <div
            className="input-group mb-3"
          >
            <span
              className="input-group-text text-white bg-secondary mr-1"
              id="basic-addon3"
            >
              https://exampleimage.com/
            </span>
            <Field
              aria-describedby="basic-addon3"
              className="form-control text-white bg-dark"
              id="url"
              name="url"
              type="text"
            />
          </div>
        </div>
        <div
          className=""
        >
          <button
            className="btn text-white w-100 mb-2 bg-dark border"
            type="submit"
          >
            Submit
          </button>
        </div>
      </Form>
      <br />
    </div>
  </Formik>
  <br />
  <br />
  <br />
</div>
`;
